src/Simplex/Chat/Types.hs

```diff
@@ -761,6 +804,16 @@ instance ToJSON GroupMemberRole where
   toJSON = strToJSON
   toEncoding = strToJEncoding
 
+data GroupMemberSettings = GroupMemberSettings
+  { showMessages :: Maybe MsgFilter
```

Comment:
for group member separate type could be used, allowing blocking, in addition to regular fields